plugins/net_plugin/net_plugin.cpp

```diff
@@ -2208,7 +2234,10 @@ namespace eosio {
          return;
       }
 
-      if( sync_state != lib_catchup || !sync_recently_active()) {
+      stages current_sync_state = sync_state;
```

Comment:
why this new variable?

Reply:
Because it is logged and it is an atomic.

Reply:
This is not done consistently, not [here]( https://github.com/AntelopeIO/spring/blob/dab3534bdb5202a8ce288eaaded4c137456dc91a/plugins/net_plugin/net_plugin.cpp#L2012-L2019) for example